### ISSUE TITLE
Log an info-level message when Zebra reaches each network upgrade

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#970]
## Motivation

After Zebra finishes verifying checkpoints, the logs are a bit boring and repetitive. It would be nice to show some progress indicators, but there's nothing interesting about each extra 10,000 blocks.

Network upgrade activation blocks are interesting though. And logging them could be useful for diagnosing any issues around network upgrades.

This is a good task for a new contributor.

### Before Verification

At the start of `ChainVerifier::call`, log an info-level message each time a network upgrade block is received. We want to log this message before starting verification, to diagnose errors or hangs that happen at network upgrade activation.

Network upgrade activation blocks are the first block in each network upgrade. You can find them using
`NetworkUpgrade::is_activation_height(network, height)`.

### After State Commit

We could also log a message after the block is committed to the state, in the `update_progress` task. If the current `ChainTip`'s network upgrade is different from the previous one, log a message.

(This task doesn't see all block heights, so we have to do the check differently.)

**Comment by teor2345** [comment]
This would be nice but it's not needed yet.

**Comment by teor2345** [comment]
We did something like this in PR #4694 - we log the current network upgrade in progress logs.
(But we don't log on the first block that activates the network upgrade.)
